src/style-spec.ts

```diff
@@ -69,20 +69,40 @@ import latest from './reference/latest';
 import format from './format';
 import migrate from './migrate';
 import derefLayers from './deref';
-import diff from './diff';
+import diff, {operations as diffOperations} from './diff';
 import ValidationError from './error/validation_error';
 import ParsingError from './error/parsing_error';
-import {StyleExpression, isExpression, createExpression, createPropertyExpression, normalizePropertyExpression, ZoomConstantExpression, ZoomDependentExpression, StylePropertyFunction} from './expression';
+import {FeatureState, StyleExpression, isExpression, createExpression, createPropertyExpression, normalizePropertyExpression, ZoomConstantExpression, ZoomDependentExpression, StylePropertyFunction, Feature, GlobalProperties, SourceExpression, CompositeExpression, StylePropertyExpression} from './expression';
 import featureFilter, {isExpressionFilter} from './feature_filter';
 
 import convertFilter from './feature_filter/convert';
 import Color from './util/color';
 import Padding from './util/padding';
+import Formatted, {FormattedSection} from './expression/types/formatted';
 import {createFunction, isFunction} from './function';
 import convertFunction from './function/convert';
 import {eachSource, eachLayer, eachProperty} from './visit';
-
+import ResolvedImage from './expression/types/resolved_image';
 import validate from './validate_style';
+import {supportsPropertyExpression} from './util/properties';
+import {number} from './util/interpolate';
+import {IMercatorCoordinate, ICanonicalTileID, ILngLat, ILngLatLike} from './tiles_and_coordinates';
+import EvaluationContext from './expression/evaluation_context';
+import {FormattedType, NullType, Type, toString} from './expression/types';
+
+import * as interpolate from './util/interpolate';
```

Comment:
Can we change this to something more specific? I don't like using `import * as...`...

Reply:
It's used like this - not clear to me how to resolve it... 

```ts
const interp: ((a: T, b: T, t: number) => T) = (interpolate as any)[this.specification.type];
```

Reply:
Reminds me of the other "generic" tricky case which also doesn't have a good solution yet:

https://github.com/maplibre/maplibre-gl-js/blob/045f1ad208f646c80058312887c92608bdda2dbd/build/generate-style-code.ts#L194-L200

Reply:
Good idea. I've implemented a simple factory like that